### PR TITLE
`azurerm_notification_hub` : Support new property `browser_credential`

### LINKED ISSUE #25588: Support for Azure Notification Hub browser (web push) credentials
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Browser (Web Push) is now in GA and available to set credentials via the portal. We need to be able to set these via the provider.

CleanShot 2024-04-11 at 13 53 52

### New or Affected Resource(s)/Data Source(s)

azurerm_notification_hub

### Potential Terraform Configuration

```hcl
resource "azurerm_notification_hub" "example" {
  name                = "mynotificationhub"
  namespace_name      = azurerm_notification_hub_namespace.example.name
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  browser_credential {
    subject_name       = "xxx"
    vapid_private_key      = "xxx"
    vapid_public_key        = "xxx"
  } 
}
```

### References

https://devblogs.microsoft.com/azure-notification-hubs/announcing-browser-push-ga-with-azure-notification-hubs/

### PR DESCRIPTION
**Pull request by sinbai**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

Support new property `browser_credential`.

Swagger:
https://github.com/Azure/azure-rest-api-specs/blob/abad0096677005817d2c19df2364663e5583c8fc/specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/stable/2023-09-01/notificationhubs.json#L2069

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
PASS: TestAccNotificationHub_browserCredential (619.26s)
PASS: TestAccNotificationHub_basic (588.45s)
PASS: TestAccNotificationHub_updateTag (983.93s)
PASS: TestAccNotificationHub_requiresImport (584.19s)
PASS: TestAccNotificationHubNamespaceDataSource_free (378.68s)
```

https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_NOTIFICATIONHUB/219255?buildTab=tests
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_notification_hub` - support for the `browser_credential` property [GH-25588]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #25588

### CODE REVIEW

File: internal/services/notificationhub/notification_hub_resource.go [hunk 1]
```diff
@@ -134,6 +134,32 @@ func resourceNotificationHub() *pluginsdk.Resource {
...
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"subject": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"vapid_private_key": {
```
Comment (stephybun) [inline internal/services/notificationhub/notification_hub_resource.go:149]: This should probably be marked as `Sensitive`

File: website/docs/r/notification_hub.html.markdown [hunk 2]
```diff
@@ -74,6 +76,16 @@ A `apns_credential` block contains:
 
 ---
 
+A `browser_credential` block contains:
```
Comment (stephybun) [inline website/docs/r/notification_hub.html.markdown:79]:
```suggestion
A `browser_credential` supports the following:
```

Can you please update this for `apns_credential` and `gcm_credential`

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @sinbai, please take a look at the comments left in-line.

File: website/docs/r/notification_hub.html.markdown (hunk 2 above)
Comment (sinbai) [reply website/docs/r/notification_hub.html.markdown]: Fixed.

File: internal/services/notificationhub/notification_hub_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/notificationhub/notification_hub_resource.go:149]: Fixed.

**Comment by sinbai** [comment]
> Thanks @sinbai, please take a look at the comments left in-line.

Hi @stephybun thanks for your time and feedback. I have updated the code, could you please take another look?

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 👍

**Pull request merged**